### DISCUSSION TITLE
ERROR: Exception occured while parsing: list index out of range

### DISCUSSION THREAD

**Discussion by olivierkeke** [mak-gitdev/HA_enoceanmqtt#111] [General]
Hi @mak-gitdev,

After one or two weeks, enocean message reception stopped. The log show a lot of lines (around 8 per seconds) `ERROR: Exception occured while parsing: list index out of range`. I can always send enocean commands after that but I receive no more measurement.

I am using the docker image `makgitdev/ha_enoceanmqtt_dev-aarch64` but I'm not sure this is the last version (`docker image ls` show that the image was created 6 month ago...)

Do I need to update to last image before submitting an issue?

**Comment by mak-gitdev** [comment]
Hi @olivierkeke,

This error has been fixed long time ago.
And the latest version has been updated 25 days ago:
![image](https://github.com/mak-gitdev/HA_enoceanmqtt/assets/73297795/8eee8330-01b6-445b-ae04-b439f7b18b34)

**Reply by olivierkeke** [reply]
Thanks! I 've updated to the last image, no problem so far.
It seems that [my build](https://hub.docker.com/r/hollysaiqs/ha_enoceanmqtt/tags) (based on the dev branch) from a month ago was affected. When was the fix added to the dev branch?

**Reply by mak-gitdev** [reply]
Hi @olivierkeke, 

This should have been fixed by version 0.1.26.
Perhaps the dockerfile was outdated at the time you built your image.

Please let me know if you still get this issue with the latest image.

**Reply by olivierkeke** [reply]
Hi,

My Dockerfile:

```
FROM python:3-alpine

ARG BUILD_VERSION

# Create config volume
VOLUME /config

# Create Working Directory
WORKDIR /app

# Install enoceanmqtt and requirements
RUN apk add --no-cache git
RUN python3 -m pip install --upgrade pip && \
    pip3 install pyyaml && \
    pip3 install tinydb && \
    pip3 install git+https://github.com/mak-gitdev/enocean.git && \
    pip3 install git+https://github.com/embyt/enocean-mqtt.git

COPY HA_enoceanmqtt HA_enoceanmqtt/

# Set entrypoint
ENTRYPOINT [ "python", "-m", "HA_enoceanmqtt.main", "/config/enoceanmqtt.conf" ]
```

Am I missing something?

**Comment by mak-gitdev** [comment]
Hi @olivierkeke,

The issue comes from the EnOcean library and from your Dockerfile, this is correctly installed into the Docker container.
So I think it should have been ok.

As previously said, let's see if this issue happens again with the latest image.

**Reply by olivierkeke** [reply]
Hi @mak-gitdev,

After two month of testing my patch without seeing this issue anymore, I've submitted a PR for this issue: https://github.com/mak-gitdev/enocean/pull/3

You can find a test image on dockerhub:
```shell
docker pull hollysaiqs/ha_enoceanmqtt:full-patches-latest
```
